source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -62,6 +67,20 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::HeaderMap& headers, b
     default_port = 443;
   }
 
+  // Check for per route filter config.
+  const std::string& name = HttpFilterNames::get().DynamicForwardProxy;
+  const ProxyPerRouteConfig* route_config =
+      route_entry->perFilterConfigTyped<ProxyPerRouteConfig>(name);
+  const ProxyPerRouteConfig* config =
+      route_config ? route_config
+                   : route_entry->virtualHost().perFilterConfigTyped<ProxyPerRouteConfig>(name);
```

Comment:
I feel like there is some utility function for this pattern. Can you take a look?

Reply:
couldn't find one, so i added one